homeassistant/components/http/view.py

```diff
@@ -103,8 +103,11 @@ def request_handler_factory(view, handler):
 
     async def handle(request):
         """Handle incoming request."""
-        if not request.app[KEY_HASS].is_running:
-            return web.Response(status=503)
+
+        # We do not want to serve http requests when
+        # stopping.
+        if not request.app[KEY_HASS].is_setup:
```

Comment:
Why do we need a `setup` phase? We can start serving once the `frontend` integration is loaded or when `EVENT_HOMEASSISTANT_START` is fired, whichever comes first.

Starting the HTTP server early is going to be interesting. I think that we can do without a blocking spinner or a reload, since we're subscription based.

Reply:
Only generateLovelaceUI might be impacted, but only if the user is _opening_ the frontend during startup. If it was already open, it would just show a bunch of "entity not found" but that's fine.

Reply:
I needed a way to know that `frontend` has finished loaded in order to start the webserver as everything breaks if the frontend isn't fully loaded because `IndexView` and other are missing. Is there an event fired when `frontend` is finished loading ?

Reply:
```
2020-05-24 21:15:17 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event component_loaded[L]: component=frontend>
```

Answering my own question here.

Reply:
Removed the setup phase.